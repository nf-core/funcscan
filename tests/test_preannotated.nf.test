nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test_preannotated"

    test("test_preannotated_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_funcscan_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                // AMPir
                { assert snapshot(
                    path("$outputDir/amp/ampir/sample_1/sample_1.ampir.tsv").text.contains("MRWGYPLSLVLMALSVAAPMIYFRRKGWLR"),
                    path("$outputDir/amp/ampir/sample_1/sample_1.ampir.faa"),
                    path("$outputDir/amp/ampir/sample_2/sample_2.ampir.tsv").text.contains("MRWGYPLSLVLMALSVAAPMIYFRRKGWLR"),
                    path("$outputDir/amp/ampir/sample_2/sample_2.ampir.faa"),
                    path("$outputDir/amp/ampir/sample_3/sample_3.ampir.tsv").text.contains("IPELEMRWGYPLSLVLMALSVAAPMIYFRRKGWLR"),
                    path("$outputDir/amp/ampir/sample_3/sample_3.ampir.faa").text.contains(">NODE_882919_length_258_cov_0.935961_1 # 149 # 256 # -1 # ID=")
                ).match("ampir") },

                // AMPlify
                { assert snapshot(
                    path("$outputDir/amp/amplify/sample_1/sample_1.amplify.tsv").text.contains("MRWGYPLSLVLMALSVAAPMIYFRRKGWLR"),
                    path("$outputDir/amp/amplify/sample_2/sample_2.amplify.tsv").text.contains("MRWGYPLSLVLMALSVAAPMIYFRRKGWLR"),
                    path("$outputDir/amp/amplify/sample_3/sample_3.amplify.tsv").text.contains("IPELEMRWGYPLSLVLMALSVAAPMIYFRRKGWLR")
                ).match("amplify") },

                // HMMsearch
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_1/sample_1_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2/sample_2_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_3/sample_3_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },

                // Macrel
                { assert snapshot(
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_3.macrel/sample_3.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_3.macrel/sample_3.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_3.macrel/sample_3.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_2.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_3.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel_log.txt"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel_log.txt"),
                    path("$outputDir/amp/macrel/sample_3.macrel/sample_3.macrel_log.txt")
                ).match("macrel") },

                // AMPcombi
                { assert snapshot(
                    path("$outputDir/reports/ampcombi2/Ampcombi_cluster.log"),
                    path("$outputDir/reports/ampcombi2/Ampcombi_complete.log"),
                    path("$outputDir/reports/ampcombi2/Ampcombi_summary.tsv").text.contains("NODE_515831_length_303_cov_1.532258_1"),
                    path("$outputDir/reports/ampcombi2/Ampcombi_parse_tables.log").text.contains(" \$\$\$\$\$\$\\  \$\$\\      \$\$\\")
                ).match("ampcombi") },

                // DeepARG
                { assert snapshot(
                    path("$outputDir/arg/deeparg/sample_1/sample_1.align.daa.tsv"),
                    path("$outputDir/arg/deeparg/sample_2/sample_2.align.daa.tsv"),
                    path("$outputDir/arg/deeparg/sample_3/sample_3.align.daa.tsv"),
                    path("$outputDir/arg/deeparg/sample_1/sample_1.mapping.ARG").text.contains("#ARG"),
                    path("$outputDir/arg/deeparg/sample_2/sample_2.mapping.ARG").text.contains("#ARG"),
                    path("$outputDir/arg/deeparg/sample_3/sample_3.mapping.ARG").text.contains("#ARG")
                ).match("deeparg") },
                { assert file("$outputDir/arg/deeparg/sample_1/sample_1.align.daa").name },
                { assert file("$outputDir/arg/deeparg/sample_2/sample_2.align.daa").name },
                { assert file("$outputDir/arg/deeparg/sample_3/sample_3.align.daa").name },
                { assert path("$outputDir/arg/deeparg/sample_1/sample_1.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_2/sample_2.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_3/sample_3.mapping.potential.ARG").text.contains("#ARG") },

                // ABRicate
                { assert snapshot(
                    path("$outputDir/arg/abricate/sample_1/sample_1.txt"),
                    path("$outputDir/arg/abricate/sample_2/sample_2.txt"),
                    path("$outputDir/arg/abricate/sample_3/sample_3.txt")
                ).match("abricate") },

                // AMRFinderPlus
                { assert snapshot(
                    path("$outputDir/arg/amrfinderplus/sample_1/sample_1.tsv"),
                    path("$outputDir/arg/amrfinderplus/sample_2/sample_2.tsv"),
                    path("$outputDir/arg/amrfinderplus/sample_3/sample_3.tsv")
                ).match("amrfinderplus") },

                // RGI
                { assert snapshot(
                    path("$outputDir/arg/rgi/sample_1/sample_1.txt"),
                    path("$outputDir/arg/rgi/sample_2/sample_2.txt"),
                    path("$outputDir/arg/rgi/sample_3/sample_3.txt")
                ).match("rgi") },

                // fARGene
                { assert snapshot(
                    path("$outputDir/arg/fargene/sample_1/class_a/results_summary.txt").text.contains("The used HMM-model was: class_A.hmm"),
                    path("$outputDir/arg/fargene/sample_2/class_a/results_summary.txt").text.contains("The used HMM-model was: class_A.hmm"),
                    path("$outputDir/arg/fargene/sample_3/class_a/results_summary.txt").text.contains("The used HMM-model was: class_A.hmm"),
                    path("$outputDir/arg/fargene/sample_1/class_b_1_2/results_summary.txt").text.contains("The used HMM-model was: class_B_1_2.hmm"),
                    path("$outputDir/arg/fargene/sample_2/class_b_1_2/results_summary.txt").text.contains("The used HMM-model was: class_B_1_2.hmm"),
                    path("$outputDir/arg/fargene/sample_3/class_b_1_2/results_summary.txt").text.contains("The used HMM-model was: class_B_1_2.hmm"),
                    path("$outputDir/arg/fargene/sample_1/fargene_analysis.log").text.contains("fARGene is done."),
                    path("$outputDir/arg/fargene/sample_2/fargene_analysis.log").text.contains("fARGene is done."),
                    path("$outputDir/arg/fargene/sample_3/fargene_analysis.log").text.contains("fARGene is done.")
                ).match("fargene") },

                // hAMRonization
                { assert snapshot(path("$outputDir/reports/hamronization_summarize/hamronization_combined_report.tsv").readLines().size()).match("hamronization") },

                // argNorm
                { assert snapshot(
                    path("$outputDir/arg/argnorm/amrfinderplus/sample_1.normalized.tsv"),
                    path("$outputDir/arg/argnorm/amrfinderplus/sample_2.normalized.tsv"),
                    path("$outputDir/arg/argnorm/amrfinderplus/sample_3.normalized.tsv")
                ).match("argnorm_amrfinderplus") },

                { assert snapshot(
                    path("$outputDir/arg/argnorm/deeparg/sample_1.ARG.normalized.tsv"),
                    path("$outputDir/arg/argnorm/deeparg/sample_1.potential.ARG.normalized.tsv"),
                    path("$outputDir/arg/argnorm/deeparg/sample_2.ARG.normalized.tsv"),
                    path("$outputDir/arg/argnorm/deeparg/sample_2.potential.ARG.normalized.tsv"),
                    path("$outputDir/arg/argnorm/deeparg/sample_3.ARG.normalized.tsv"),
                    path("$outputDir/arg/argnorm/deeparg/sample_3.potential.ARG.normalized.tsv")
                ).match("argnorm_deeparg") },

                { assert snapshot(
                    path("$outputDir/arg/argnorm/abricate/sample_1.normalized.tsv"),
                    path("$outputDir/arg/argnorm/abricate/sample_2.normalized.tsv"),
                    path("$outputDir/arg/argnorm/abricate/sample_3.normalized.tsv")
                ).match("argnorm_abricate") }
            )
        }
    }
}
