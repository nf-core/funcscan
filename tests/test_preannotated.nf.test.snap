{
    "deeparg": {
        "content": [
            "sample_1.align.daa.tsv:md5,0e71c37318bdc6cba792196d0455293d",
            "sample_2.align.daa.tsv:md5,1092ecd3cd6931653168b46c7afeb9e3",
            "sample_3.align.daa.tsv:md5,a9ed2f0651d75b318fde07a76b06d7b8",
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:24.751995878"
    },
    "ampir": {
        "content": [
            true,
            "sample_1.ampir.faa:md5,ab02c6e9c5f36ba9c31af97f95f9c317",
            true,
            "sample_2.ampir.faa:md5,12826875bd18623da78770187a7bbd2c",
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-20T11:01:38.808336691"
    },
    "argnorm_amrfinderplus": {
        "content": [
            "sample_1.normalized.tsv:md5,0a7f76ceb606ac46730a51dd57290768",
            "sample_2.normalized.tsv:md5,602afce3ee0ee179855c848bd87208fe",
            "sample_3.normalized.tsv:md5,d4fb8fbd890217eb4d667d7a4dd80c9b"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:25.55764618"
    },
    "argnorm_abricate": {
        "content": [
            "sample_1.normalized.tsv:md5,ddd8d454672c57b798f477ca32504a42",
            "sample_2.normalized.tsv:md5,0323fc890a8f698ac4b0ac25f5e65964",
            "sample_3.normalized.tsv:md5,f71490c27790071bd5974ecc5502cf73"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:23:32.486921338"
    },
    "amplify": {
        "content": [
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:24.483855968"
    },
    "argnorm_deeparg": {
        "content": [
            "sample_1.ARG.normalized.tsv:md5,26aa409bfd0fc9096f2ac404760cc492",
            "sample_1.potential.ARG.normalized.tsv:md5,d6732b4b9765bfa47e27ba673e24b6a4",
            "sample_2.ARG.normalized.tsv:md5,1a19b894a7315aaae5f799e4539e6619",
            "sample_2.potential.ARG.normalized.tsv:md5,b241e22f9116d8f518ba8526d52ac4dc",
            "sample_3.ARG.normalized.tsv:md5,d7577c0066a31e173f9cb545820650bf",
            "sample_3.potential.ARG.normalized.tsv:md5,6d0889215f0ad7f601502ca67c0ca89e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:23:32.446555281"
    },
    "macrel": {
        "content": [
            "sample_1.macrel.smorfs.faa.gz:md5,a4f853b560c6a8c215e0d243c24ec056",
            "sample_2.macrel.smorfs.faa.gz:md5,83ae7b9808d7183d87b41c10253c9c9e",
            "sample_3.macrel.smorfs.faa.gz:md5,a4f853b560c6a8c215e0d243c24ec056",
            "sample_1.macrel.all_orfs.faa.gz:md5,d1ae1cadc3770994b2ed4982aadd5406",
            "sample_2.macrel.all_orfs.faa.gz:md5,d9612a4275a912cabdae13b1ccc1857e",
            "sample_3.macrel.all_orfs.faa.gz:md5,d1ae1cadc3770994b2ed4982aadd5406",
            "sample_1.macrel.prediction.gz:md5,62146cf9f759c9c6c2c2f9e5ba816119",
            "sample_2.macrel.prediction.gz:md5,1b479d31bb7dbf636a2028ddef72f5cc",
            "sample_3.macrel.prediction.gz:md5,62146cf9f759c9c6c2c2f9e5ba816119",
            "README.md:md5,cf088d9256ff7b7730699f17b64b4028",
            "README.md:md5,cf088d9256ff7b7730699f17b64b4028",
            "README.md:md5,cf088d9256ff7b7730699f17b64b4028",
            "sample_1.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_2.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_3.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T11:50:30.926088397"
    },
    "hamronization": {
        "content": [
            246
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-05T10:17:06.711064611"
    },
    "abricate": {
        "content": [
            "sample_1.txt:md5,427cec26e354ac6b0ab6047ec6621202",
            "sample_2.txt:md5,4c140c932a48a22bcd8ae911bda8f4c7",
            "sample_3.txt:md5,d6534efe3d03173749d003bf9e624e68"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:24.87794287"
    },
    "fargene": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:25.248986515"
    },
    "rgi": {
        "content": [
            "sample_1.txt:md5,ff8f179d06d8566d8cf779fc7d1f4955",
            "sample_2.txt:md5,cc4ae1fb9e0d5f79ef5105d640c7b748",
            "sample_3.txt:md5,ff8f179d06d8566d8cf779fc7d1f4955"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:25.117843821"
    },
    "ampcombi": {
        "content": [
            "Ampcombi_cluster.log:md5,4c78f5f134edf566f39e04e3ab7d8558",
            "Ampcombi_complete.log:md5,3dabfea4303bf94bd4f5d78c5b8c83c1",
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:24.639509225"
    },
    "amrfinderplus": {
        "content": [
            "sample_1.tsv:md5,29cfb6f34f420d802eda95c6d9daa361",
            "sample_2.tsv:md5,d9b6565167d603a1f07cff2374db8eb2",
            "sample_3.tsv:md5,29cfb6f34f420d802eda95c6d9daa361"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-27T08:11:24.994284774"
    }
}